### PR TITLE
Update brew extension

### PR DESCRIPTION
**Pull request by ashokbaruaakas**
## Description

Added Brew Clean Up Command in existing Brew Extension

## Screencast

image
image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/brew/package.json [hunk 1]
```diff
@@ -39,6 +40,13 @@
       "subtitle": "Brew",
       "description": "Upgrade brew formulae & casks",
       "mode": "no-view"
+    },
+    {
+      "name": "clean-up",
+      "title": "Clean Up",
+      "subtitle": "Brew",
+      "description": "Clean Up Everything At Once",
```
Comment (nhojb) [inline extensions/brew/package.json:48]: Better just use "Sentence case" here (uppercase just the first word) 😄

File: extensions/brew/src/brew.ts [hunk 2]
```diff
@@ -293,6 +293,10 @@ export async function brewUpgradeAll(greedy: boolean, cancel?: AbortController):
   await execBrew(cmd, cancel);
 }
 
+export async function brewCleanup(cancel?: AbortController): Promise<void> {
+  await execBrew(`cleanup --prune=all`, cancel);
```
[the remark this answers is not in the document]
Comment (nhojb) [reply extensions/brew/src/brew.ts:297]: Here are the developer docs related to command properties and preferences:

https://developers.raycast.com/information/manifest#command-properties

[the remark this answers is not in the document]
Comment (ashokbaruaakas) [reply extensions/brew/src/brew.ts]: Yes... Thanks for informing and i already updated my code for that.
[Commit](https://github.com/raycast/extensions/pull/10704/commits/75abaea8ccd589e07b8031dd38eb15c8fed7e47a)

File: extensions/brew/package.json (hunk 1 above)
Comment (ashokbaruaakas) [reply extensions/brew/package.json]: Thanks... & I fix that

File: extensions/brew/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Brew Changelog
 
+## [Upgrades] - 2024-02-11
```
[the remark this answers is not in the document]
Comment (ashokbaruaakas) [reply extensions/brew/CHANGELOG.md]: Yes... Thanks for informing and i already updated my code for that.

File: extensions/brew/src/clean-up.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { showToast, Toast } from "@raycast/api";
+import { brewCleanup } from "./brew";
+import { showActionToast, showFailureToast, wait } from "./utils";
+
+export default async (): Promise<void> => {
+  try {
+    const abort = showActionToast({ title: "Cleaning up everything at once" + String.ellipsis, cancelable: true });
```
[the remark this answers is not in the document]
Comment (ashokbaruaakas) [reply extensions/brew/src/clean-up.tsx]: Yes... Thanks for informing and i already updated my code for that.

**Comment by ashokbaruaakas** [comment]
> Thanks for the contribution 👏
> 
> Generally looks good. The only thing I think would be good is to add a command preference to enable the `--prune=all` behaviour. Disabled by default.

Thanks for informing and i already updated my code for that.

**Review by nhojb — CHANGES_REQUESTED** [review]
Thanks for the changes!

One last thing: the new preference should be associated with the `clean-up` command, rather than the extension. You just need a new `preferences` entry in the command's JSON definition.

**Comment by ashokbaruaakas** [comment]
> Thanks for the changes!
> 
> One last thing: the new preference should be associated with the `clean-up` command, rather than the extension. You just need a new `preferences` entry in the command's JSON definition.

Oh.. Thanks Again & It's Done...

**Comment by pernielsentikaer** [comment]
Thanks for the corrections @ashokbaruaakas - I changed a bit to get preferences automatically 🙂

@nhojb can you approve this?

**Comment by ashokbaruaakas** [comment]
> Thanks for the corrections @ashokbaruaakas - I changed a bit to get preferences automatically 🙂
> 
> @nhojb can you approve this?

Thank you for making those adjustments! As I'm relatively new here,  I've learned from your feedback...

Thanks again! ❤️

**Comment by pernielsentikaer** [comment]
Why did you add `getPreferenceValues` back @ashokbaruaakas :)

**Comment by ashokbaruaakas** [comment]
Apologies for the confusion. I recently pulled the changes, resolved conflicts, and tested them locally. Unfortunately, it seems I mistakenly added that function. I'll rectify this error promptly. Thank you for your understanding. @pernielsentikaer

**Comment by pernielsentikaer** [comment]
No worries at all, was just curious 🙂

**Comment by ashokbaruaakas** [comment]
> Thanks for the changes!
> 
> One last thing: the new preference should be associated with the `clean-up` command, rather than the extension. You just need a new `preferences` entry in the command's JSON definition.

Upon reviewing the previous configuration preferences of this extension, I believe it would be better to incorporate this preference as an extension preference. @nhojb 

image

**Comment by nhojb** [comment]
@ashokbaruaakas the reason the "Upgrades", "Uninstall" and "Quarantine" preferences are on the extension is that they can apply to multiple commands. For example there's an `Upgrade` command and an `Outdated` command which both can upgrade formulae. Similarly you can uninstall from the `Search` or `Show Installed` command.

With cleanup the `--prune` option only applies to that command, so to me it makes more sense to place it there.

**Comment by ashokbaruaakas** [comment]
@nhojb  I've made the changes we discussed. Please review now. Thank You...

**Review by nhojb — APPROVED** [review]
Looks good. Thanks for your patience 😄

**Comment by ashokbaruaakas** [comment]
> 🎉 🎉 🎉
> 
> Such a great contribution deserves a reward, but unfortunately we couldn't find your Raycast account based on your GitHub username (@ashokbaruaakas). Please [link your GitHub account to your Raycast account](https://www.raycast.com/settings/account) to receive your credits and soon be able to exchange them for some swag.

Thank you so much! I truly appreciate the recognition from Raycast. I've just linked my Raycast account with GitHub. Looking forward to exploring the rewards. 🎉

**Pull request merged**